Rakefile

```diff
@@ -86,4 +86,4 @@ namespace :test do
   end
 end
 
-task :default => [:rubocop, "test:all"]
+task :default => [*(:rubocop if RUBY_ENGINE == "ruby"), "test:all"]
```

Comment:
What does RUBY_ENGINE return on TruffleRuby/JRuby?

I guess this is fine, maybe a bit annoying though if you're working on Puma using JRuby and then get surprised by CI failing b/c of whitespace.

Reply:
`RUBY_ENGINE` is simply `"truffleruby"` and `"jruby"` there.
I was tempted to replace `defined?(JRUBY_VERSION)` with `RUBY_ENGINE == "jruby"` in the codebase since that's the portable way to test the Ruby implementation, but avoided so in this PR to keep it smaller.

Given RuboCop runs (from what I can see locally) significantly slower (> 10s) on both of these implementations I think it's likely to be less annoying if it doesn't run on those implementations.

I could do that change only in the CI file, but I thought it's actually helpful also when running `bundle exec rake` locally.

Reply:
I think we should probably keep it the same for all implementations. If you personally think RC is too slow, you can still just run `rake test:all`.

Reply:
OK, I'll move that check to the CI file, won't be as nice though.

Reply:
Done!